TileMap.gd

```diff
@@ -528,3 +536,29 @@ func get_required_and_free_doors_to_palce(doorsToAdd, forcedDoorsToPlace, curren
 		set_tile(currentDirectionMidPoint.x, currentDirectionMidPoint.y, directionWallNeeded[workingDirection])
 	
 	return [doorsToAdd, forcedDoorsToPlace]
+
+func place_treasure_chest(anchorPosition):
+	#generatePossible chest spawnPoints
+	var wallAndDoorOffset = 2
+	#remove spots where wall are	
+	#remove spots where a door could be
```

Comment:
Does this account for possibly placing it on the stairs/ladder?

Reply:
yes, because stair/ladder only spawns in a possible starting spot